src/pages/workspace/WorkspaceSettingsCurrencyPage.js

```diff
@@ -1,78 +1,48 @@
 
 const propTypes = {
-    /** Constant, list of available currencies */
-    currencyList: PropTypes.objectOf(
-        PropTypes.shape({
-            /** Symbol of the currency */
-            symbol: PropTypes.string.isRequired,
-        }),
-    ),
     isLoadingReportData: PropTypes.bool,
     ...policyPropTypes,
 };
 
 const defaultProps = {
-    currencyList: {},
+    /** Route from navigation */
+    route: PropTypes.shape({
+        /** Params from the route */
+        params: PropTypes.shape({
+            /** Focused currency code */
+            currency: PropTypes.string,
+
+            /** ID of a policy */
+            policyID: PropTypes.string,
+        }),
+    }).isRequired,
     isLoadingReportData: true,
     ...policyDefaultProps,
 };
 
-const getDisplayText = (currencyCode, currencySymbol) => `${currencyCode} - ${currencySymbol}`;
-
-function WorkspaceSettingsCurrencyPage({currencyList, policy, isLoadingReportData}) {
+function WorkspaceSettingsCurrencyPage({route, policy, isLoadingReportData}) {
     const {translate} = useLocalize();
-    const [searchText, setSearchText] = useState('');
-    const trimmedText = searchText.trim().toLowerCase();
-    const currencyListKeys = _.keys(currencyList);
-
-    const filteredItems = _.filter(currencyListKeys, (currencyCode) => {
-        const currency = currencyList[currencyCode];
-        return getDisplayText(currencyCode, currency.symbol).toLowerCase().includes(trimmedText);
-    });
-
-    let initiallyFocusedOptionKey;
-
-    const currencyItems = _.map(filteredItems, (currencyCode) => {
-        const currency = currencyList[currencyCode];
-        const isSelected = policy.outputCurrency === currencyCode;
-
-        if (isSelected) {
-            initiallyFocusedOptionKey = currencyCode;
-        }
-
-        return {
-            text: getDisplayText(currencyCode, currency.symbol),
-            keyForList: currencyCode,
-            isSelected,
-        };
-    });
-
-    const sections = [{data: currencyItems, indexOffset: 0}];
-
-    const headerMessage = searchText.trim() && !currencyItems.length ? translate('common.noResultsFound') : '';
-
-    const onBackButtonPress = useCallback(() => Navigation.goBack(ROUTES.WORKSPACE_SETTINGS.getRoute(policy.id)), [policy.id]);
+    const currencyParam = lodashGet(route, 'params.currency', '').toUpperCase();
+    const selectedCurrencyCode = CurrencyUtils.isValidCurrencyCode(currencyParam) ? currencyParam : policy.outputCurrency;
+    const onBackButtonPress = useCallback(() => Navigation.navigate(ROUTES.WORKSPACE_SETTINGS.getRoute(policy.id, selectedCurrencyCode)), [policy.id, selectedCurrencyCode]);
 
     const onSelectCurrency = (item) => {
-        Policy.updateGeneralSettings(policy.id, policy.name, item.keyForList);
-        Navigation.goBack(ROUTES.WORKSPACE_SETTINGS.getRoute(policy.id));
+        Navigation.navigate(ROUTES.WORKSPACE_SETTINGS.getRoute(policy.id, item.currencyCode));
```

Comment:
I am using navigate here because, goBack won't properly pass currency parameter when user select a currency to previous page. 

Country selection page also use navigate:

https://github.com/Expensify/App/blob/1f72cef0a19a312e085255f101aba0f93a478619/src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.js#L70


The video:

https://github.com/Expensify/App/assets/114975543/06ada993-c1d1-41c4-941a-e9915ded1a5d